src/main/java/Main.kt

```diff
@@ -30,7 +30,12 @@ object Main {
 
         //
         val rootFunctionModule = createVisitorFunction(ASTRoot)
-        rootFunctionModule.invoke(targetTreeNode)
+        val functionResult = rootFunctionModule.invoke(targetTreeNode)
+
+        if (functionResult.hasMatch)
+            println(targetTreeNode.matchedNodesString(functionResult.matches))
+        else
+            println("Match not found")
```

Comment:
I'm not sure if we are supposed to remove the prints or it can be left like this. If we want to get rid of them, this could work:
```suggestion
        val solution: String
        if (functionResult.hasMatch)
            solution = targetTreeNode.matchedNodesString(functionResult.matches)
        else
            solution = "Match not found"
```
And in case we want to make full use of kotlin, this could be done:
```suggestion
        val solution: String
        solution = if (functionResult.hasMatch)
            targetTreeNode.matchedNodesString(functionResult.matches)
        else
            "Match not found"
```

Reply:
Yes, is better save first in a var, maybe later print that but doesn't matter